vscode-extension/src/extension.ts

```diff
@@ -26,14 +39,82 @@ export function activate(context: vscode.ExtensionContext) {
   });
   context.subscriptions.push(setupCommand);
 
+  let createAIConfigJSONCommand = vscode.commands.registerCommand(
+    COMMANDS.CREATE_NEW_JSON,
+    async () => {
+      return await createNewAIConfig(context, aiconfigEditorManager, "json");
+    }
+  );
+  context.subscriptions.push(createAIConfigJSONCommand);
+
+  let createAIConfigYAMLCommand = vscode.commands.registerCommand(
+    COMMANDS.CREATE_NEW_YAML,
+    async () => {
+      return await createNewAIConfig(context, aiconfigEditorManager, "yaml");
+    }
+  );
+  context.subscriptions.push(createAIConfigYAMLCommand);
+
+  let shareModelParserCommand = vscode.commands.registerCommand(
+    COMMANDS.SHARE,
+    async () => {
+      return await shareAIConfig(context, aiconfigEditorManager);
+    }
+  );
+  context.subscriptions.push(shareModelParserCommand);
+
+  let customModelParserCommand = vscode.commands.registerCommand(
+    COMMANDS.CUSTOM_MODEL_REGISTRY_PATH,
+    async () => {
+      return await registerCustomModelRegistry(aiconfigEditorManager);
+    }
+  );
+  context.subscriptions.push(customModelParserCommand);
+
+  let createCustomModelRegistryCommand = vscode.commands.registerCommand(
+    COMMANDS.CREATE_CUSTOM_MODEL_REGISTRY,
+    async () => {
+      return await createCustomModelRegistry(context, aiconfigEditorManager);
+    }
+  );
+  context.subscriptions.push(createCustomModelRegistryCommand);
+
+  let openModelParserCommand = vscode.commands.registerCommand(
+    COMMANDS.OPEN_MODEL_REGISTRY,
+    async () => {
+      return await openModelRegistry(context, aiconfigEditorManager);
+    }
+  );
+  context.subscriptions.push(openModelParserCommand);
+
   // Run the setup command on activation
   vscode.commands.executeCommand(COMMANDS.INIT);
 
-  vscode.window.showInformationMessage("Hello World from aiconfig-editor!");
-
   // Register our custom editor providers
+  const aiconfigEditorManager: AIConfigEditorManager =
+    new AIConfigEditorManager();
+
   context.subscriptions.push(
-    AIConfigEditorProvider.register(context, extensionOutputChannel)
+    AIConfigEditorProvider.register(
+      context,
+      extensionOutputChannel,
+      aiconfigEditorManager
+    )
+  );
+
+  // Also handle file renames/moves -- inform the EditorManager of the change
+  context.subscriptions.push(
+    vscode.workspace.onDidRenameFiles((e) => {
+      e.files.forEach(async (file) => {
+        const editor = aiconfigEditorManager.getEditorByUri(
+          file.oldUri.toString()
+        );
+        if (editor) {
+          aiconfigEditorManager.removeEditorByUri(file.oldUri.toString());
+          aiconfigEditorManager.addEditor(editor, file.newUri.toString());
```

Comment:
If this is renaming a file whose editor is not the active one in view, won't this incorrectly make that new editor active?
Maybe we should move this into a `replaceEditor` method in the manager which also handles setting the proper active editor if the rename happens on the active one?